be/src/exec/olap_scanner.cpp

```diff
@@ -538,6 +538,14 @@ Status OlapScanner::close(RuntimeState* state) {
     _reader.reset();
     Expr::close(_conjunct_ctxs, state);
     _is_closed = true;
+    if (config::scan_count_push_tablet_into_compaction_heap != 0 &&
+        _tablet->query_scan_count->value() % config::scan_count_push_tablet_into_compaction_heap ==
+                0) {
+        _tablet->data_dir()->push_tablet_into_compaction_heap(CompactionType::BASE_COMPACTION,
```

Comment:
Have you test the performance of `push_tablet_into_compaction_heap`.
This is on the critical path of READ, and the default `scan_count_push_tablet_into_compaction_heap` is 1,
so the tablet will call this method every time it being read.

If the query concurrency is high, i doubt it may impact the performance.

Reply:
@morningman 
The default max size of compaction heap is 20. After test, the cost of `push_tablet_into_compaction_heap`for each tablet is `0.05 ms` averagely.

Reply:
The default `scan_count_push_tablet_into_compaction_heap` is 1 and this value can be set dynamically.